src/go/rpk/pkg/cli/transform/logs.go

```diff
@@ -0,0 +1,523 @@
+* by the Apache License, Version 2.0
+ */
+
+package transform
+
+import (
+	"container/list"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"os/signal"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"go.uber.org/zap"
+)
+
+var (
+	minTime = time.Unix(0, 0).UTC()
+	maxTime = time.Unix(math.MaxInt64, math.MaxInt64).UTC()
+)
+
+func newLogsCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var follow bool
+	var head, tail int
+	var since, until timequery
+	since.time = minTime
+	until.time = maxTime
+	cmd := &cobra.Command{
+		Use:     "logs NAME",
+		Aliases: []string{"log"},
+		Short:   "View logs for a transform",
+		Long: `View logs for a transform.
+
+Data transform's STDOUT and STDERR are captured during runtime and written to a topic.
+This command allows viewing logs for a single transform over a period of time and printing them
+to STDOUT. The logs can be printed in various formats.
+
+The --follow flag allows for streaming logs continuously, by default only emitted logs are displayed.
+
+FILTERING
+
+The --head and --tail flags are mutually exclusive and allow for limiting the number of log entries
+from the beginning and end of the range, respectively.
+
+The --since and --until flags allow for limiting the logs to a period of time. 
+The flag's value can be provided in a few different options:
+
+    
+    now                   the current time, useful for --since=now
+    13 digits             parsed as a unix millisecond
+    9 digits              parsed as a unix second
+    YYYY-MM-DD            parsed as a day, UTC
+    YYYY-MM-DDTHH:MM:SSZ  parsed as RFC3339, UTC; fractional seconds optional (.MMM)
+    -dur                  a negative duration from now
+    dur                   a positive duration from now
+
+Durations are parsed simply:
+
+    3ms    three milliseconds
+    10s    ten seconds
+    9m     nine minutes
+    1h     one hour
+    1m3ms  one minute and three milliseconds
+
+For example,
+
+    --since=-1h   reads logs within the last hour
+    --until=-30m  reads logs prior to 30 minutes ago
+
+FORMATTING
+
+Logs can be displayed in a variety of formats using --format.
+
+The default --format=text prints the log record's body line by line.
+
+When --format=wide is specified includes a prefix that is the date of the 
+log line and a level for the record. INFO level corresponds to being emitted
+on the transform's STDOUT, while the WARN level is used for STDERR.
+
+The --format=json flag emits logs in their raw format, which is
+the JSON encoded version of the Open Telemetry LogRecord 
+protocol buffer.`,
```

Comment:
not sure if saying that logs are in their raw format adds much info. @kbatuigas thoughts? 

```suggestion
The --format=json flag emits logs in the JSON encoded version of the Open Telemetry LogRecord 
protocol buffer.`,
```

Reply:
Yeah, probably OK to leave it out

Reply:
It is relevant per that comment, but this probably isn't the right place to document that. I'll drop